### PR TITLE
Enable deploying api operator in any namespace

### PR DESCRIPTION
**Pull request by pubudu538**
## Purpose
- Operator runs on any deployed namespace by default.
- Operator can be run as cluster scope after providing relevant roles and config changes.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: api-operator/pkg/config/system.go [hunk 1]
```diff
@@ -43,3 +45,31 @@ func SetOperatorNamespace() {
 		OperatorNamespace = ns
 	}
 }
+
+// Checks whether operator is used cluster wide or namespace wide.
+func getWatchClusterLevel() (watchClusterLevel bool) {
```
Comment (renuka-fernando) [inline api-operator/pkg/config/system.go:50]: Shall we start the Godoc comment with the function name? the convention for Godoc is to start the comment with its declaration,
Ref: https://blog.golang.org/godoc

File: api-operator/deploy/controller-configs/api_controller_conf.yaml [hunk 2]
```diff
@@ -84,7 +84,7 @@ kind: ConfigMap
 metadata:
   name: envoy-mgw-configs
 data:
-  mgwAdapterHost: "https://adapter.default:9843"
+  mgwAdapterHost: "https://adapter-service:9843"
```
Comment (renuka-fernando) [inline api-operator/deploy/controller-configs/api_controller_conf.yaml:87]: From MG m4 release we changed the name of the adapter service to `adapter` to align with docker-compose service name
https://github.com/wso2/product-microgateway/pull/1574/files#diff-7e2af33631f6c25b77eb84c52fccc822592cfd958dc435d4994e5164f9a8ca84

Comment (pubudu538) [reply api-operator/deploy/controller-configs/api_controller_conf.yaml:87]: But still, we support M3 and this is the working config.

File: api-operator/pkg/config/system.go (hunk 1 above)
Comment (pubudu538) [reply api-operator/pkg/config/system.go]: Fixed. Thank you for pointing this.

File: api-operator/deploy/controller-configs/api_controller_conf.yaml (hunk 2 above)
Comment (renuka-fernando) [reply api-operator/deploy/controller-configs/api_controller_conf.yaml:87]: go it

File: api-operator/pkg/config/system.go [hunk 3]
```diff
@@ -43,3 +45,31 @@ func SetOperatorNamespace() {
...
+	if watchClusterLevel {
+		ns, _ := k8sutil.GetWatchNamespace()
+		if ns != "" {
+			watchNamespaces = ns
+		} else {
+			watchNamespaces = ""
+		}
+
+	} else {
+		watchNamespaces = OperatorNamespace
```
Comment (renuka-fernando) [inline api-operator/pkg/config/system.go:71]: shall we check `OperatorNamespace` and `SystemNamespace` here, and if not equals panic execution?

**Review by renuka-fernando — APPROVED** [review]

**Pull request merged**